src/protocol/frame/frame.rs

```diff
@@ -304,7 +289,7 @@ impl Frame {
             0 => Ok(None),
             1 => Err(Error::Protocol(ProtocolError::InvalidCloseSequence)),
             _ => {
-                let mut data = self.payload.as_slice();
+                let mut data = BytesMut::from(self.payload);
```

Comment:
This isn't ideal (though no worse than before), maybe we should also convert `CloseFrame::reason` to bytes?

Reply:
I made that change